src/RazorSdk/SourceGenerators/MetadataReferenceExtensions.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.NET.Sdk.Razor.SourceGenerators
+{
+    internal static class MetadataReferenceExtensions
+    {
+        public static Guid? GetModuleVersionId(this MetadataReference reference, Compilation compilation)
+        {
+            var symbol = compilation.GetAssemblyOrModuleSymbol(reference);
+
+            if (symbol is IAssemblySymbol assemblySymbol)
+            {
+                foreach (var module in assemblySymbol.Modules)
```

Comment:
I don't believe that we have any guarantee that this list is ordered. Hence executing this multiple times could produce different results. Consider sorting this list to make the return consistent. 

Even better, I would just bail if the count of `Modules` here was greater than one. That means you're dealing with a multi-module assembly which is **extremely** rare and not even really supported on .NET Core. Not caching here isn't likely going to have any observable impact on our dev loops.